packages/loader/container-loader/src/container.ts

```diff
@@ -946,6 +946,79 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         }
     }
 
+    public connect() {
+        if (!this.closed && !this.connected) {
+            // Note: no need to fetch ops as we do it preemptively as part of DeltaManager.attachOpHandler().
+            // If there is gap, we will learn about it once connected, but the gap should be small (if any),
+            // assuming that connect() is called quickly after initial container boot.
+            this.connectInternal({ reason: "DocumentOpenResume", fetchOpsFromStorage: false });
+        }
+    }
+
+    private connectInternal(args: IConnectionArgs) {
+        assert(!this.closed, "Attempting to connect() a closed DeltaManager");
+        assert(this._attachState === AttachState.Attached, "Attempting to connect() a container that is not attached");
```

Comment:
Can you switch this to throw a UsageError?  Or probably better to just add that check to `connect` then leave this an assert (since it's private)

Reply:
Currently if you call `connect()` on a closed container it is a no-op. Should calling `connect()` on a detached container also be a no-op, or do you think we should add a UsageError for both of these cases in `connect()`?

Reply:
Let's consider updating BREAKING.md since this is a behavior change a partner needs to consider when switching from `setAutoReconnect(true)` to `connect()` (they'll need to do the checks if we're not gonna do them for them)